### PR TITLE
Color pie traces with colorscale

### LINKED ISSUE #538: Can't change colors of pie chart slices
I might be missing something in the [demo app](https://plotly.github.io/react-chart-editor/), but it doesn't seem that one can change the pie chart slice colors:

image

image

### PR DESCRIPTION
**Pull request by VeraZab**
Closes #538

### CODE REVIEW

File: src/lib/index.js [hunk 1]
```diff
@@ -175,7 +176,33 @@ function maybeAdjustSrc(src, srcAttributePath, traceType, config) {
...
+  config
+) {
+  if (config.repeat) {
+    const repetitions = Math.ceil(numberOfNeededColors / colorscale.length);
+    const newArray = new Array(repetitions).fill(colorscale);
+    return newArray
+      .reduce((a, b) => {
+        return a.concat(b);
+      }, [])
+      .slice(0, numberOfNeededColors);
```
Comment (dmt0) [inline src/lib/index.js]: What do you need that newArray for?

```
    return new Array(repetitions).fill(colorscale
      .reduce((a, b) => a.concat(b), [])
      .slice(0, numberOfNeededColors);
```

File: src/components/fields/Colorscale.js [hunk 2]
```diff
@@ -12,6 +12,18 @@ class Colorscale extends Component {
 
   onUpdate(colorscale, colorscaleType) {
     if (Array.isArray(colorscale)) {
+      if (this.context.container.type === 'pie') {
+        const pieSlices = this.context.container.labels.length;
```
Comment (dmt0) [inline src/components/fields/Colorscale.js]: it's more like numPieSlices?

File: src/lib/index.js (hunk 1 above)
Comment (VeraZab) [reply src/lib/index.js]: its just my way of generating an array of colors that repeat themselves enough times to cover all the traces that need to be colored..

**Comment by dmt0** [comment]
Only looked at code.
Just those two minor things, otherwise 💃!

File: src/lib/index.js (hunk 1 above)
Comment (dmt0) [reply src/lib/index.js]: yeah, I figured, just saying could be written sorter without extra consts

**Comment by nicolaskruchten** [comment]
From conversation: we should generate N colors where N is the length of calcdata, not labels.

**Comment by nicolaskruchten** [comment]
💃 !

**Pull request merged**